coreneuron/sim/treeset_core.cpp

```diff
@@ -144,6 +150,7 @@ static void nrn_lhs(NrnThread* _nt) {
            so here we transform so it only has membrane current contribution
         */
         double* p = _nt->nrn_fast_imem->nrn_sav_d;
+#pragma acc parallel loop present(p, vec_d) if (_nt->compute_gpu) async(_nt->stream_id)
```

Comment:
This is launched via `async` but I believe the synchronisation via `acc wait` is not immediately required (there should be top level `acc wait` already). cc: @iomaganaris

Reply:
That was my conclusion too; let's wait for @iomaganaris's review though.

Reply:
IIUC `stream_id` should always be `0` unless we use `openmp` threading, which in the normal case we don't. 
More details here: https://github.com/BlueBrain/CoreNeuron/blob/f61ec320b2f36fb9579006a741235f35e4b25e6e/coreneuron/io/phase2.cpp#L843
However since there is still this case I think that the `async(stream_id)` is still needed